woocommerce/woocommerce-gutenberg-products-block
# Sale Product Block: Enable global style

~This PR is blocked by #5542~

This PR adds support for global style for the `Sale Product block`. This code registers the block on the PHP side. This is necessary for using the `__experimentalSelector` from Global Styles API.


Now, the user can edit the style for:
- border-radius
- font-size
- text color
- background-color (including gradients)
- padding
- color 
- radius


#4965 

### Screenshots

![image](https://user-images.githubusercontent.com/4463174/153616647-a4368f19-f154-40c9-8885-ee5de7c2df99.png)
*without global style*


![image](https://user-images.githubusercontent.com/4463174/153219070-18f2839d-d06a-4f6e-b00e-04c3d118f653.png)

*with global style*


### Testing

### Manual Testing

Check out this branch:

1. Install and enable the `Twenty Twenty-Two` theme.
2. Add the `All Product Block` to a post.
3. Go to Dashboard and select Appearance > Editor (beta).
4. On the Editor page click on the `Styles` icon on the right-top corner.
5. Verify that the `Sale Product block` is shown under the `Blocks` section. Click on the label. Personalize the block.
6. Save your changes.
7. Go on the page created earlier and check if all styles are applied correctly.

### Changelog

> Add support for the global style for the Sale Product block.
